### PR TITLE
Finish comments feature completely.

### PR DESCRIPTION
**Pull request by sofifch**
I managed to make the comments section of my portfolio presentable in terms of aesthetics, and it is also fully functional on the part of storing and loading data. I will go on with week 3's work and come back to this feature to improve it!

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/data/Comment.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+public final class Comment {
+
+    private final String name;
+    private final String messageContent;
+
+    public Comment(String name, String messageContent) {
+        this.name = name;
+        this.messageContent = messageContent;
+    }
+}
```
Comment (rjkroege) [inline portfolio/src/main/java/com/google/sps/data/Comment.java]: add a trailing new line

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -14,7 +14,16 @@
...
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.sps.data.Comment;
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import com.google.gson.Gson;
```
Comment (rjkroege) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Please place in alphabetic order.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -26,7 +35,38 @@
...
+            String personMessage = (String) entity.getProperty("messageContent");
+            myMessages.add(personName + ": " + personMessage);
+        }
+
+        String myJson = convertToJsonWithGson(myMessages);
+        response.setContentType("application/json;");
+        response.getWriter().println(myJson);
+  }
+
+  private String convertToJsonWithGson(List<String> arraylist) {
```
Comment (rjkroege) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Can be static

File: portfolio/src/main/webapp/comments.html [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+            <input type="text" name="person-comment">
+            <br/><br/>
+            <input type="submit"/>
+        </form>
+        <div class="d-flex flex-column">
+            <p id="message-container"></p>
+        </div>
+      </div>
+  </body>
+</html>
```
Comment (rjkroege) [inline portfolio/src/main/webapp/comments.html]: Please add trailing newline

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -36,8 +36,19 @@ function myFunction() {
   document.getElementById("myBar").style.width = scrolled + "%";
 }
 
-async function getMessage() {
-    const response = await fetch('/data');
-    const message = await response.text();
-    document.getElementById('message-container').innerHTML = message;
+function getMessage() {
+    fetch('/data').then(response => response.json()).then((messageList) => {
+
```
Comment (rjkroege) [inline portfolio/src/main/webapp/script.js]: why blank line?

File: portfolio/src/main/webapp/script.js [hunk 6]
```diff
@@ -36,8 +36,19 @@ function myFunction() {
...
+        const commentsListElement = document.getElementById('message-container');
+        commentsListElement.innerHTML = '';
+        for (i = 0; i < messageList.length; i++) {
```
Comment (rjkroege) [inline portfolio/src/main/webapp/script.js:43]: FYI: You could use the `reduce` function. (But don't have to.)

File: portfolio/src/main/webapp/style2.css [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+  width: 100%;
+  height: 6px;
+  background: #ccc;
+}
+
+.progress-bar {
+  height: 6px;
+  background: #ebbde1;
+  width: 0%;
+}
```
Comment (rjkroege) [inline portfolio/src/main/webapp/style2.css]: Please add a trailing newline

File: portfolio/src/main/java/com/google/sps/data/Comment.java (hunk 1 above)
Comment (sofifch) [reply portfolio/src/main/java/com/google/sps/data/Comment.java]: Added!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (sofifch) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Fixed!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 3 above)
Comment (sofifch) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Added!

File: portfolio/src/main/webapp/comments.html (hunk 4 above)
Comment (sofifch) [reply portfolio/src/main/webapp/comments.html]: Added!

File: portfolio/src/main/webapp/script.js (hunk 5 above)
Comment (sofifch) [reply portfolio/src/main/webapp/script.js]: Removed!

File: portfolio/src/main/webapp/script.js (hunk 6 above)
Comment (sofifch) [reply portfolio/src/main/webapp/script.js:43]: I didn't know! I went over its documentation and learned about it. Thanks for the tip.

File: portfolio/src/main/webapp/style2.css (hunk 7 above)
Comment (sofifch) [reply portfolio/src/main/webapp/style2.css]: Added!

File: portfolio/pom.xml [hunk 8]
```diff
@@ -23,6 +23,16 @@
       <version>4.0.1</version>
       <scope>provided</scope>
     </dependency>
+    <dependency>
+        <groupId>com.google.code.gson</groupId>
```
Comment (rjkroege) [inline portfolio/pom.xml:27]: This is mixed in with the other CL yes?

File: portfolio/src/main/java/com/google/sps/data/Comment.java [hunk 9]
```diff
@@ -0,0 +1,12 @@
+package com.google.sps.data;
+
+public final class Comment {
+
+    private final String name;
```
Comment (rjkroege) [inline portfolio/src/main/java/com/google/sps/data/Comment.java:5]: Nice to make these immutable once created.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 10]
```diff
@@ -14,7 +14,16 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
```
Comment (rjkroege) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:17]: Excellent ordering!

**Review by rjkroege — APPROVED** [review]
This seems good to me. I'm a bit lost about what changes should be separate but this PR seems good as a stand-alone thing.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 10 above)
Comment (sofifch) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:17]: Thanks!

File: portfolio/src/main/java/com/google/sps/data/Comment.java (hunk 9 above)
Comment (sofifch) [reply portfolio/src/main/java/com/google/sps/data/Comment.java:5]: :)

File: portfolio/pom.xml (hunk 8 above)
Comment (sofifch) [reply portfolio/pom.xml:27]: yes, you are right!

**Pull request merged**